akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -14,19 +14,32 @@ import akka.util.OptionVal
 /**
  * INTERNAL API
  */
-@InternalApi private[typed] class ActorAdapter[T](_initialBehavior: Behavior[T]) extends a.Actor {
+@InternalApi private[typed] object ActorAdapter {
+  case object Start
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[typed] class ActorAdapter[T](_initialBehavior: Behavior[T], delayStart: Boolean) extends a.Actor {
   import Behavior._
   import ActorRefAdapter.toUntyped
 
   var behavior: Behavior[T] = _initialBehavior
 
-  if (!isAlive(behavior)) context.stop(self)
-
-  val ctx = new ActorContextAdapter[T](context)
+  var _ctx: ActorContextAdapter[T] = _
+  def ctx: ActorContextAdapter[T] =
+    if (_ctx ne null) _ctx
+    else throw new IllegalStateException("Context was accessed before typed actor was started.")
 
   var failures: Map[a.ActorRef, Throwable] = Map.empty
 
-  def receive = {
+  def receive = running
+
+  def waitingForStart: Receive = {
+    case ActorAdapter.Start ⇒ start()
```

Comment:
ok, so stashing is not needed because no message should be sent to it before started. The ActorRef (ActorSystem) has not been returned yet. The only case would be if some crazy (invalid) extension would send to the system when loaded eagerly. I'm fine with not supporting that.